gateways/common/webhook.go

```diff
@@ -73,16 +73,20 @@ type activeServer struct {
 	errChan chan error
 }
 
-// RouteConfig contains configuration about an http route
-type RouteConfig struct {
-	Webhook            *Webhook
-	Configs            map[string]interface{}
-	EventSource        *gateways.EventSource
-	Log                zerolog.Logger
-	StartCh            chan struct{}
-	RouteActiveHandler func(writer http.ResponseWriter, request *http.Request, rc *RouteConfig)
-	PostActivate       func(rc *RouteConfig) error
-	PostStop           func(rc *RouteConfig) error
+// Route contains common information for a route
+type Route struct {
+	Webhook     *Webhook
+	Logger      *zerolog.Logger
+	StartCh     chan struct{}
+	EventSource *gateways.EventSource
+}
+
+// RouteConfig is an interface to manage  the configuration for a route
+type RouteConfig interface {
```

Comment:
Is there any way to easily identify this `RouteConfig` is an interface and the other `RouteConfig`s defined in gateways are struct? Maybe rename this `RouteConfigInterface`?

Reply:
Not sure if we want `Interface` in the name, how about `RouteManager`

Reply:
Sound good 👍